src/Microsoft.Framework.PackageManager/Restore/LockFileWriter.cs

```diff
@@ -0,0 +1,235 @@
+            json["targets"] = WriteObject(lockFile.Targets, WriteTarget);
+            json["libraries"] = WriteObject(lockFile.Libraries, WriteLibrary);
+            json["projectFileDependencyGroups"] = WriteObject(lockFile.ProjectFileDependencyGroups, WriteProjectFileDependencyGroup);
+            return json;
+        }
+
+        private JProperty WriteLibrary(LockFileLibrary library)
+        {
+            var json = new JObject();
+            if (library.IsServiceable)
+            {
+                WriteBool(json, "serviceable", library.IsServiceable);
+            }
+            json["sha512"] = WriteString(library.Sha512);
+            WritePathArray(json, "files", library.Files, WriteString);
+            return new JProperty(
+                library.Name + "/" + library.Version.ToString(),
+                json);
+        }
+
+        private JProperty WriteTarget(LockFileTarget target)
+        {
+            var json = WriteObject(target.Libraries, WriteTargetLibrary);
+
+            var key = target.TargetFramework + (target.RuntimeIdentifier == null ? "" : "/" + target.RuntimeIdentifier);
+
+            return new JProperty(key, json);
+        }
+
+        private JProperty WriteTargetLibrary(LockFileTargetLibrary library)
+        {
+            var json = new JObject();
+
+            if (library.Dependencies.Count > 0)
+            {
+                json["dependencies"] = WriteObject(library.Dependencies, WritePackageDependency);
+            }
+
+            if (library.FrameworkAssemblies.Count > 0)
+            {
+                json["frameworkAssemblies"] = WriteArray(library.FrameworkAssemblies, WriteFrameworkAssemblyReference);
+            }
+
+            if (library.CompileTimeAssemblies.Count > 0)
+            {
+                json["compile"] = WritePathArray(library.CompileTimeAssemblies, WriteString);
+            }
+
+            if (library.RuntimeAssemblies.Count > 0)
+            {
+                json["runtime"] = WritePathArray(library.RuntimeAssemblies, WriteString);
+            }
+
+            if (library.NativeLibraries.Count > 0)
+            {
+                json["native"] = WritePathArray(library.NativeLibraries, WriteString);
+            }
+
+            return new JProperty(library.Name + "/" + library.Version, json);
+        }
+
+        private JProperty WriteProjectFileDependencyGroup(ProjectFileDependencyGroup frameworkInfo)
+        {
+            return new JProperty(
+                frameworkInfo.FrameworkName,
+                WriteArray(frameworkInfo.Dependencies, WriteString));
+        }
+
+        private JToken WriteFrameworkAssemblies(IList<FrameworkAssemblyReference> frameworkAssemblies)
+        {
+            var groups = frameworkAssemblies.SelectMany(x =>
+            {
+                if (x.SupportedFrameworks.Any())
+                {
+                    return x.SupportedFrameworks.Select(xx => new { x.AssemblyName, FrameworkName = xx });
```

Comment:
I copied the same code over there. It's not in the scope of this PR to bring sense to them :) Let's rename next time